src/sbd-cluster.c

```diff
@@ -499,7 +506,7 @@ find_pacemaker_remote(void)
 
         /* entry_name is truncated to 16 characters including the nul terminator */
         cl_log(LOG_DEBUG, "Found %s at %u", entry_name, pid);
-        if (strcmp(entry_name, "pacemaker_remot") == 0) {
+        if (strncmp(entry_name, PACEMAKER_REMOTE_BINARY, 15) == 0) {
```

Comment:
This makes me notice that there's no need for a char[64] to hold the 15-character name in process_info(), no idea why I did that. :) It would be nice if there were a public kernel API constant for that length, but there's not. Also, I used %s to scan the name, so it can't handle spaces in the name, though of course that's not an issue here.

Reply:
Looks as if the format-string is the wrong way round. I'm seeing keys that are way beyond 15 chars. Of course what you are doing isn't dangerous though as it just takes the first 15 chars and that doesn't write beyond the end of key and it is enough to identify "Name:".
We could use "Name:\t%15c" and fully rely on scan (no key checking). Of course zeroing out the char[16] first shouldn't be forgotten.

Reply:
Good idea, I'm thinking: "Name:\t%15[^\n]" and name[15]=0

Reply:
That is better, yes. Guess it even should write the terminating '\0' in all cases. Maybe something like "Name:\t[a-zA-Z0-9 _-]" would be enough already and avoids reading garbage. (Just the kernel worker-threads have '/'.)